agent/azure-monitor-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/AzureMonitorTraceExporter.java

```diff
@@ -129,72 +152,94 @@ public final class AzureMonitorTraceExporter implements SpanExporter {
             .build();
   }
 
-  private final TelemetryItemExporter telemetryItemExporter;
-  private final String instrumentationKey;
+  private final boolean captureHttpServer4xxAsError;
+  private final Consumer<AbstractTelemetryBuilder> defaultsPopulator;
+  private final BiPredicate<EventData, String> eventSuppressor;
+  private final Function<List<TelemetryItem>, CompletableResultCode> exportFn;
+  private final Supplier<CompletableResultCode> flushFn;
+  private final Supplier<CompletableResultCode> shutdownFn;
+  private final Supplier<Boolean> active;
+  private final Supplier<String> appIdSupplier;
 
   /**
    * Creates an instance of exporter that is configured with given exporter client that sends
    * telemetry events to Application Insights resource identified by the instrumentation key.
-   *
-   * @param httpPipeline The http pipeline used to send data to Azure Monitor.
-   * @param endpoint the ingestion endpoint used to send data to Azure Monitor.
-   * @param instrumentationKey The instrumentation key of Application Insights resource.
    */
-  AzureMonitorTraceExporter(HttpPipeline httpPipeline, URL endpoint, String instrumentationKey) {
-    TelemetryPipeline pipeline = new TelemetryPipeline(httpPipeline, endpoint);
-
-    File tempDir =
-        TempDirs.getApplicationInsightsTempDir(
-            LoggerFactory.getLogger(AzureMonitorTraceExporter.class),
-            "Telemetry will not be stored to disk and retried later"
-                + " on sporadic network failures");
-
-    if (tempDir != null) {
-      telemetryItemExporter =
-          new TelemetryItemExporter(
-              pipeline,
-              new LocalStorageTelemetryPipelineListener(
-                  TempDirs.getSubDir(tempDir, "telemetry"),
-                  pipeline,
-                  LocalStorageStats.noop(),
-                  false));
-    } else {
-      telemetryItemExporter = new TelemetryItemExporter(pipeline, TelemetryPipelineListener.noop());
-    }
-    this.instrumentationKey = instrumentationKey;
+  public AzureMonitorTraceExporter(
```

Comment:
AzureMonitorSpanExporter? logData will go to trace table and this confused me a bit? SpanData will take care of other entities except metrics and logs.

Reply:
I think this name was decided across languages because it's "trace" signal in OTel terminology. I'm ok to rename, but we should coordinate with other languages.